tests/unit/Trace/Reporter/ZipkinReporterTest.php

```diff
@@ -97,11 +91,36 @@ public function testSpanKind()
         };
 
         $this->assertCount(5, $spans);
-        // default unknown spans to client send/receive
-        $this->assertEquals(['cs', 'cr'], array_map($annotationValue, $spans[0]['annotations']));
-        $this->assertEquals(['cs', 'cr'], array_map($annotationValue, $spans[1]['annotations']));
-        $this->assertEquals(['sr', 'ss'], array_map($annotationValue, $spans[2]['annotations']));
-        $this->assertEquals(['ms'], array_map($annotationValue, $spans[3]['annotations']));
-        $this->assertEquals(['mr'], array_map($annotationValue, $spans[4]['annotations']));
+        $this->assertFalse(array_key_exists('kind', $spans[0]));
+        $this->assertEquals('CLIENT', $spans[1]['kind']);
+        $this->assertEquals('SERVER', $spans[2]['kind']);
+        $this->assertEquals('PRODUCER', $spans[3]['kind']);
+        $this->assertEquals('CONSUMER', $spans[4]['kind']);
+    }
+
+    public function testSpanDebug()
+    {
+        $tracer = new ContextTracer(new TraceContext('testtraceid'));
+        $tracer->inSpan(['name' => 'main'], function () {});
+
+        $reporter = new ZipkinReporter('myapp', 'localhost', 9411);
+        $spans = $reporter->convertSpans($tracer, [
+            'HTTP_X_B3_FLAGS' => '1'
+        ]);
+
+        $this->assertCount(1, $spans);
+        $this->assertTrue($spans[0]['debug']);
+    }
+
+    public function testSpanShared()
+    {
+        $tracer = new ContextTracer(new TraceContext('testtraceid', 12345));
+        $tracer->inSpan(['name' => 'main'], function () {});
+
+        $reporter = new ZipkinReporter('myapp', 'localhost', 9411);
+        $spans = $reporter->convertSpans($tracer);
+
+        $this->assertCount(1, $spans);
+        $this->assertTrue($spans[0]['shared']);
```

Comment:
This test provides an initial `traceId` (testtraceid) and `spanId` (12345) to the `ContextTracer`. We assume if that root span has a parent id, then it was provided from a parent caller and that this request is part of a shared trace.